ecosystem/typescript/sdk/examples/typescript/your_coin.ts

```diff
@@ -0,0 +1,122 @@
+    const rawTxn = await this.generateTransaction(coinReceiver.address(), {
+      function: "0x1::managed_coin::register",
+      type_arguments: [`${coinTypeAddress.hex()}::moon_coin::MoonCoin`],
+      arguments: [],
+    });
+
+    const bcsTxn = await this.signTransaction(coinReceiver, rawTxn);
+    const pendingTxn = await this.submitTransaction(bcsTxn);
+
+    return pendingTxn.hash;
+  }
+
+  /** Mints the newly created coin to a specified receiver address */
+  async mintCoin(minter: AptosAccount, receiverAddress: HexString, amount: number | bigint): Promise<string> {
+    const rawTxn = await this.generateTransaction(minter.address(), {
+      function: "0x1::managed_coin::mint",
+      type_arguments: [`${minter.address()}::moon_coin::MoonCoin`],
+      arguments: [receiverAddress.hex(), amount],
+    });
+
+    const bcsTxn = await this.signTransaction(minter, rawTxn);
+    const pendingTxn = await this.submitTransaction(bcsTxn);
+
+    return pendingTxn.hash;
+  }
+
+  /** Return the balance of the newly created coin */
+  async getBalance(accountAddress: MaybeHexString, coinTypeAddress: HexString): Promise<string | number> {
+    try {
+      const resource = await this.getAccountResource(
+        accountAddress,
+        `0x1::coin::CoinStore<${coinTypeAddress.hex()}::moon_coin::MoonCoin>`,
+      );
+
+      return parseInt((resource.data as any)["coin"]["value"]);
+    } catch (_) {
+      return 0;
+    }
+  }
+}
+
+/** run our demo! */
+async function main() {
+  assert(process.argv.length == 3, "Expecting an argument that points to the moon_coin directory.");
+
+  const client = new CoinClient();
+  const faucetClient = new FaucetClient(NODE_URL, FAUCET_URL);
+
+  // Create two accounts, Alice and Bob, and fund Alice but not Bob
+  const alice = new AptosAccount();
+  const bob = new AptosAccount();
+
+  console.log("\n=== Addresses ===");
+  console.log(`Alice: ${alice.address()}`);
+  console.log(`Bob: ${bob.address()}`);
+
+  await faucetClient.fundAccount(alice.address(), 20_000);
+  await faucetClient.fundAccount(bob.address(), 20_000);
+
+  await new Promise<void>((resolve) => {
+    readline.question("Update the module with Alice's address, compile, and press enter.", () => {
+      resolve();
+      readline.close();
+    });
+  });
+
+  // :!:>publish
+  const modulePath = process.argv[2];
+  const packageMetadata = fs.readFileSync(path.join(modulePath, "build", "Examples", "package-metadata.bcs"));
+  const moduleData = fs.readFileSync(path.join(modulePath, "build", "Examples", "bytecode_modules", "moon_coin.mv"));
+
+  console.log("Publishing MoonCoin package.");
+  let txnHash = await client.publishPackage(alice, new HexString(packageMetadata.toString("hex")).toUint8Array(), [
+    new TxnBuilderTypes.Module(new HexString(moduleData.toString("hex")).toUint8Array()),
+  ]);
+  await client.waitForTransaction(txnHash, { checkSuccess: true });
+  // <:!:publish
```

Comment:
You have to put this at the end of line 106, otherwise the code blocks have a trailing newline. Can we do this please? This is what the other examples do.